File: 1524-string-matching-in-an-array/string-matching-in-an-array.py
class Solution:
    def stringMatching(self, words: List[str]) -> List[str]:
        result = []

        for i in range(len(words)):
            for j in range(len(words)):
                if (words[i] in words[j]) and (words[i] != words[j]):
                    if words[i] in result:
                        continue
                    result.append(words[i])
        return result

File: 1524-string-matching-in-an-array/string-matching-in-an-array.java
class Solution {
    public List<String> stringMatching(String[] words) {
        ArrayList<String> result = new ArrayList<String>();
        for (int i = 0; i < words.length ; i++) {
            for (int j = 0; j < words.length ; j++) {
                if (words[j].contains(words[i]) && !words[j].equals(words[i])) {
                    if (!result.contains(words[i])){
                        result.add(words[i]);
                    }
                }
            }
        }
        return result;
    }
}